integration/tests/basic/rpc/test_rpc_base_calls.py

```diff
@@ -0,0 +1,650 @@
+        )
+        assert "error" not in response
+        assert "result" in response, AssertMessage.DOES_NOT_CONTAIN_RESULT
+        result = response["result"]
+        assert_fields_are_hex(result, [
+            "transactionHash",
+            "transactionIndex",
+            "blockNumber",
+            "blockHash",
+            "cumulativeGasUsed",
+            "gasUsed",
+            "logsBloom",
+            "status",
+        ])
+        assert result["status"] == "0x1", "Transaction status must be 0x1"
+        assert result["transactionHash"] == transaction_hash
+        assert result["blockHash"] == tx_receipt.blockHash.hex()
+        assert result["from"].upper() == tx_receipt["from"].upper()
+        assert result["to"].upper() == tx_receipt["to"].upper()
+        assert result["contractAddress"] is None
+        assert result["logs"] == []
+
+    def test_eth_get_transaction_receipt_when_hash_doesnt_exist(self):
+        """Verify implemented rpc calls work eth_getTransactionReceipt when transaction hash doesn't exist"""
+        response = self.proxy_api.send_rpc(
+            method="eth_getTransactionReceipt", params=gen_hash_of_block(32)
+        )
+        assert response["result"] is None, "Result should be None"
+
+    def test_eth_block_number(self):
+        """Verify implemented rpc calls work work eth_blockNumber"""
+        response = self.proxy_api.send_rpc(method="eth_blockNumber")
+        assert "error" not in response
+        assert rpc_checks.is_hex(response["result"]), f"Invalid response result {response['result']}"
+
+    def test_eth_block_number_next_block_different(self):
+        response = self.proxy_api.send_rpc(method="eth_blockNumber")
+        time.sleep(1)
+        response2 = self.proxy_api.send_rpc(method="eth_blockNumber")
+
+        assert "error" not in response and response2
+        assert rpc_checks.is_hex(response["result"]), f"Invalid response result {response['result']}"
+        assert rpc_checks.is_hex(response2["result"]), f"Invalid response result {response2['result']}"
+        assert response['result'] != response2['result']
+
+    @pytest.mark.parametrize("param", [Tag.LATEST, Tag.PENDING, Tag.EARLIEST, Tag.SAFE, Tag.FINALIZED, None])
+    def test_eth_get_storage_at(self, event_caller_contract, param: tp.Union[Tag, None]):
+        """Verify implemented rpc calls work eht_getStorageAt"""
+        response = self.proxy_api.send_rpc(
+            method="eth_getStorageAt",
+            params=[event_caller_contract.address, hex(1), param.value]
+            if param
+            else param,
+        )
+        if not param:
+            assert "error" in response, "Error not in response"
+            return
+        assert "error" not in response
+        result = response["result"]
+        assert rpc_checks.is_hex(result), f"Invalid response: {result}"
+        assert int(result, 16) != 0, "expected that result is not 0, but got 0"
```

Comment:
why you compare with 0? You know value in this case

Reply:
Changed to precise value

Reply:
http://neon-test-allure.s3-website.eu-central-1.amazonaws.com/basic/night-stand/152/merge/920/#categories/bc6cdfead4d9df5e06323c5d14635ccd/b5e1337458658af6/

Returned this case to > 0 check, as the value in ci and locally is different. Possibly due to compiler difference, arm/intel.